Copy loleaflet-api.html into code: defaultZoom

paradoxicals opened the pull request:
Copy loleaflet-api.html into code: defaultZoom as said in #2160

paradoxicals commented:
Hello @pedropintosilva,

Even if I did signed commit, the workflows(DCO ) were failed. What am I doing wrong?

mrkara commented:
Hi @paradoxicals ! You are missing signature on 1 commit, and you are missing Change-Id on all commits. If you click the Details link next to the failed check, you will get instructions on how to fix the issue.

mrkara commented:
Hey @paradoxicals ! This is my suggestion to fix this PR:

* Squash your 3 commits into 1: https://www.internalpointers.com/post/squash-commits-into-one-git
* Sign your commit if not already signed: `git commit --amend --no-edit --signoff`
* Add change-id to your commit:
`cp .git-hooks/commit-msg .git/hooks/commit-msg `
` git commit --amend `
* Force-push your changes: `git push -f`

Please don't hesitate to ping me or @pedropintosilva if you get stuck at any step. Looking forward to getting this PR merged! :)

paradoxicals commented:
> Hey @paradoxicals ! This is my suggestion to fix this PR:
> 
> * Squash your 3 commits into 1: https://www.internalpointers.com/post/squash-commits-into-one-git
> * Sign your commit if not already signed: `git commit --amend --no-edit --signoff`
> * Add change-id to your commit:
>   `cp .git-hooks/commit-msg .git/hooks/commit-msg `
>   `git commit --amend`
> * Force-push your changes: `git push -f`
> 
> Please don't hesitate to ping me or @pedropintosilva if you get stuck at any step. Looking forward to getting this PR merged! :)

Hello @mrkara ,

I tried to squash my commits on gitpod. I hope it will work this time. 

I have already signed off my commits before but I did the other steps also.
![Ekran Alıntısı](https://user-images.githubusercontent.com/2785064/125245762-9df8e280-e2f9-11eb-92cb-8ef7aa69e03b.PNG)

mrkara commented:
@paradoxicals Oh, you now have 4 commits, with an additional `merge` commit on top of the previous 3. :)

mrkara commented:
@paradoxicals Here is another alternative: If you can go to https://github.com/paradoxicals/online/settings/access and invite me as a collaborator, and give me maintainer-level access to your fork, I can help to iron this out.

paradoxicals commented:
@mrkara  I think I am lost between the platforms, sorry about that. I have sent an invitation to you.

mrkara commented:
> @mrkara I think I am lost between the platforms, sorry about that. I have sent an invitation to you.

@paradoxicals Hey! No worries. We will fix this. ;)

I've checked my notifications, and also the invites section of your fork, but couldn't find anything. Could you please make sure you have invited `mrkara` (my username)?

mrkara commented:
@paradoxicals I got the invitation, and squashed all your commits into one, added the Change-Id, and also signed it. But my signature is redundant. Now only missing piece is your signature. So:

* Go to gitpod and delete all workspaces
* Go to your fork: https://github.com/paradoxicals/online/
* Click on the Gitpod button to start a new Gitpod env
* Issue this command to sign the last commit: `git commit --amend -s`
* Push the change: `git push -f`

paradoxicals commented:
> @paradoxicals I got the invitation, and squashed all your commits into one, added the Change-Id, and also signed it. But my signature is redundant. Now only missing piece is your signature. So:
> 
> * Go to gitpod and delete all workspaces
> * Go to your fork: https://github.com/paradoxicals/online/
> * Click on the Gitpod button to start a new Gitpod env
> * Issue this command to sign the last commit: `git commit --amend -s`
> * Push the change: `git push -f`

@mrkara I have finished all steps above right now

mrkara commented:
@paradoxicals You again have an unwanted merge commit. Have you done `anything` in addition to the steps above?

paradoxicals commented:
> @paradoxicals You again have an unwanted merge commit. Have you done `anything` in addition to the steps above? -Maybe clicked some buttons on GitHub? :)-

@mrkara Actually I did all steps(which you wrote) in Gitpod enviroment .. one pull request should not be that hard or confusing :) I dont know what I did wrong.. Can I delete this pull request?

pedropintosilva commented:
@paradoxicals it seems you have two commits :

https://github.com/CollaboraOnline/online/pull/2830/commits

1. Copy loleaflet-api.html into code: defaultZoom
2. Merge branch 'CollaboraOnline:master' into master (which has many commits in it)

The second one (the last) is actually the same but it's without commit message, so we should get rid of it. The first is the one we want but it has two signatures so we need to remove one (from @mrkara )

so the best why to do this is to rebase it interactively like so

1.
```bash
git rebase -i HEAD~2
```
it will open the text editor
and you will see:
2.
```bash
pick 8a5c91acf Copy loleaflet-api.html into code: defaultZoom
pick f8bea2e9c HTTP 302 Found - redirrect support for getWOPIFileInfo
pick 850f3cee7 HTTP 302 Found - remember the correct resource address
pick 3330293e5 HTTP 302 Found - redirrect support for downloadDocument
pick b72c8e9ae Handle all redirect HTTP verbs
pick fef43e237 HTTP 302 Found - block redirection loops
pick 27799f23e HTTP 302 - updates after review
pick c61bcca1c Rename fileId to redirectionCount
pick 016244c1f Leaflet-progress: use proper css vars
pick e823c8d40 cypress: fix: apply_paragraph_props_shape/text fails due to core change
```
and you need to delete the `pick` words and replace it with `d` (drop) and `r` to reword  like so
```bash
r 8a5c91acf Copy loleaflet-api.html into code: defaultZoom
d f8bea2e9c HTTP 302 Found - redirrect support for getWOPIFileInfo
d 850f3cee7 HTTP 302 Found - remember the correct resource address
d 3330293e5 HTTP 302 Found - redirrect support for downloadDocument
d b72c8e9ae Handle all redirect HTTP verbs
d fef43e237 HTTP 302 Found - block redirection loops
d 27799f23e HTTP 302 - updates after review
d c61bcca1c Rename fileId to redirectionCount
d 016244c1f Leaflet-progress: use proper css vars
d e823c8d40 cypress: fix: apply_paragraph_props_shape/text fails due to core changes
```

pedropintosilva reviewed (approved):
Well done : ),  I will merge it once the tests are done

paradoxicals commented:
> Well done : ), I will merge it once the tests are done

@pedropintosilva Thanks a lot :) So for this PR, the other tests are not important, are they?

For the next PRs I should commit with -s option for signing.
How should I be certain about change-id ? Should I copy every time this file (cp .git-hooks/commit-msg .git/hooks/commit-msg) before commit ?

pedropintosilva commented:
> For the next PRs I should commit with -s option for signing.

Yes

> How should I be certain about change-id ?

no, it will automatically generate for you

> Should I copy every time this file (cp .git-hooks/commit-msg .git/hooks/commit-msg) before commit ?

I do not use GitPod but, and if it's similar to how it works locally, I think that step is only needed once (?) maybe @kara can confirm this

paradoxicals commented:
> > For the next PRs I should commit with -s option for signing.
> 
> Yes
> 
> > How should I be certain about change-id ?
> 
> no, it will automatically generate for you
> 
> > Should I copy every time this file (cp .git-hooks/commit-msg .git/hooks/commit-msg) before commit ?
> 
> I do not use GitPod but, and if it's similar to how it works locally, I think that step is only needed once (?) maybe @kara can confirm this

@pedropintosilva Thank you for detailed answers. 

@mrkara  on Gitpod env, should I copy every time this file (cp .git-hooks/commit-msg .git/hooks/commit-msg) before commit ?

pedropintosilva commented:
@paradoxicals also please be careful with the use of mixed spaces and tabs. I completely miss that test and thought it passed but it didn't, see : https://github.com/CollaboraOnline/online/runs/3046624916#step:6:171 

For this PR you don't need to worry, it was already merged a fix for those mixed spaces and tabs : https://github.com/CollaboraOnline/online/commit/5f46edc83931af8e315ce65445f2e4770dcf299b

Generally speaking you need to be sure you can see what your editor is inserting when you hit `tab` on your keyboard if spaces or tabs. For CSS and JS it should be always tabs

mrkara commented:
> > > For the next PRs I should commit with -s option for signing.
> > 
> > 
> > Yes
> > > How should I be certain about change-id ?
> > 
> > 
> > no, it will automatically generate for you
> > > Should I copy every time this file (cp .git-hooks/commit-msg .git/hooks/commit-msg) before commit ?
> > 
> > 
> > I do not use GitPod but, and if it's similar to how it works locally, I think that step is only needed once (?) maybe @kara can confirm this
> 
> @pedropintosilva Thank you for detailed answers.
> 
> @mrkara on Gitpod env, should I copy every time this file (cp .git-hooks/commit-msg .git/hooks/commit-msg) before commit ?

Hey @paradoxicals !

You should never need to do that. If you create/open a Gitpod instance from the main repo (https://github.com/CollaboraOnline/online), it comes prebuilt with everything set up.

If you create/open a Gitpod instance from your fork, then you will need to build it first, which again makes sure everything is in place.

You need to manually copy the git hooks only when the above options fail for some reason.

pedropintosilva commented:
@paradoxicals so you need to be sure that your editor :

- Shows those formatting characters (so you can see what it really inserted)
- Inserts a tab and not a space 

in VS Code (which I think is what gitpod uses by default) you can go press `ctrl` + `,` (File, Preferences, settings) search for `renderWhitespace` and change its value to `all`:
![image](https://user-images.githubusercontent.com/65948705/125407523-40789a80-e3ba-11eb-8530-b2a999e16992.png)

then close and open again GitPod and you will now see the difference in every file between spaces and tabs:
![image](https://user-images.githubusercontent.com/65948705/125408721-7bc79900-e3bb-11eb-8b61-23bf7ec3a97a.png)

----

Avoid inserting spaces by mistaken when tabbing: File > Preferences > Settings `insertSpaces`:
![image](https://user-images.githubusercontent.com/65948705/125409063-d2cd6e00-e3bb-11eb-9776-940bc0163abb.png)
